service/service_validate.go

```diff
@@ -82,6 +86,59 @@ func ValidateService(service *Service) error {
 	return errs.ErrorOrNil()
 }
 
+// isServiceKeysUnique checks uniqueness of service deps/tasks/events/params keys.
+func isServiceKeysUnique(service *Service) error {
+	var errs xerrors.Errors
+	exist := make(map[string]bool)
+	for _, dep := range service.Dependencies {
+		if exist[dep.Key] {
+			errs = append(errs, fmt.Errorf("dependencies[%s] already exist", dep.Key))
+		}
+		exist[dep.Key] = true
+	}
+
+	exist = make(map[string]bool)
+	for _, task := range service.Tasks {
+		if exist[task.Key] {
+			errs = append(errs, fmt.Errorf("tasks[%s] already exist", task.Key))
+		}
+		exist[task.Key] = true
+
+		existparam := make(map[string]bool)
+		for _, param := range task.Inputs {
+			if existparam[param.Key] {
+				errs = append(errs, fmt.Errorf("tasks[%s].inputs[%s] already exist", task.Key, param.Key))
+			}
+			existparam[param.Key] = true
+		}
+
+		existparam = make(map[string]bool)
+		for _, param := range task.Outputs {
+			if existparam[param.Key] {
+				errs = append(errs, fmt.Errorf("tasks[%s].outputs[%s] already exist", task.Key, param.Key))
+			}
+			existparam[param.Key] = true
+		}
+	}
+
+	exist = make(map[string]bool)
+	for _, event := range service.Events {
+		if exist[event.Key] {
+			errs = append(errs, fmt.Errorf("events[%s] already exist", event.Key))
+		}
+		exist[event.Key] = true
+
+		existparam := make(map[string]bool)
+		for _, param := range event.Data {
+			if existparam[param.Key] {
```

Comment:
params.Object keys are not tested (task inputs, task outputs and event)
maybe we could use the validation lib with something like:
```
Object []*Parameter `hash:"name:7" validate:"dive,(key)unique,required"`
```

Reply:
otherwise a new custom validation like `PortMapping`